fakes_test.go

```diff
@@ -7,13 +7,51 @@
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4/fake"
 	"github.com/stretchr/testify/require"
 )
 
+// matthchr: I don't like that this has to be package-scoped and not scoped to the individual test. There's gonna be a proliferation of these in test files.
+type fakeGetServer struct {
+	fake.VirtualMachinesServer
+
+	getVMResponse armcompute.VirtualMachine
+}
+
+// matthchr: No autocompletion of method name or signature, which is annoying, because no way for me to signify that I want to override a particular method. I need to go
+// copy the method from fake.VirtualMachinesServer. Obviously I can do that it'd be nicer if I could intellisense it. Especially because when I copy it I also have to go fix up
+// the imports for Responder[...] and ErrorResponder to be fakeazcore.Responder and fakeazcore.ErrorResponder.
+// matthchr: Actually, I can't replace with fakeazcore.Responder and fakeazcore.ErrorResponder, I need to replace with fake.Responder and fakeazcore.Responder... why are there two copies?
+// The error was:
+// have Get(ctx context.Context, resourceGroupName string, vmName string, options *armcompute.VirtualMachinesClientGetOptions) (resp "github.com/Azure/azure-sdk-for-go/sdk/azcore/fake".Responder[armcompute.VirtualMachinesClientGetResponse], err "github.com/Azure/azure-sdk-for-go/sdk/azcore/fake".ErrorResponder)
+// want Get(ctx context.Context, resourceGroupName string, vmName string, options *armcompute.VirtualMachinesClientGetOptions) (resp "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4/fake".Responder[armcompute.VirtualMachinesClientGetResponse], err "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4/fake".ErrorResponder)
+func (s *fakeGetServer) Get(ctx context.Context, resourceGroupName string, vmName string, options *armcompute.VirtualMachinesClientGetOptions) (resp fake.Responder[armcompute.VirtualMachinesClientGetResponse], err fake.ErrorResponder) {
+	// matthchr: Curious why not let me set the response and/or error in the construction of the fakeazcore.Responder?
+	// it's pretty minor but IMO this is a bit cleaner as it's a single return statement. I'm not sure if Go generics is smart enough to infer the [T]
+	// in the construction but if it is that's nice:
+	// return fakeazcore.Responder[armcompute.VirtualMachinesClientGetResponse]{
+	//    Resp: armcompute.VirtualMachinesClientGetResponse{
+	//		VirtualMachine: s.vm,
+	//	}
+	// }, fakeazcore.ErrorResponder{}
+	// matthchr: I do see for the errorResponder it lets you do errResponder.SetResponseError, which is nice. Maybe just matching that pattern for the standard responder too?
+	// matthchr: Ok after using it more, I definitely don't mind the way it's done. It's most awkward for the ErrorResponder and Responder, for the PagerResponder and PollerResonder it works nicer
+	responder := fake.Responder[armcompute.VirtualMachinesClientGetResponse]{}
```

Comment:
You are correct that generics aren't smart enough at present to infer the type.  This is why I used named return variables, in a hope to simplify populating the responses.  Just cut/paste then call the appropriate `Set` method.